include/kernel.h

```diff
@@ -14,7 +14,7 @@
 #define ZEPHYR_INCLUDE_KERNEL_H_
 
 #if !defined(_ASMLANGUAGE)
-#include <kernel_includes.h>
+#include <private/kernel_includes.h>
```

Comment:
do we intent to work on it in future ? It's really weird include private headers in a public header.

Reply:
that is  actually a lie, kernel_includes.h is not private, it just includes many other headers and should not be included directly. This will be changed.